tests/src/Rules/data/intentionally-unused-throws.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types = 1);
+
+namespace Pepakriz\PHPStanExceptionRules\Rules\UselessThrows;
+
+use RuntimeException;
+
+interface Foo {
+
+	/**
+	 * @throws RuntimeException
+	 */
+	public function method(): void;
+
+}
+
+class ImplementationExample implements Foo
+{
+
+	public function method(): void
+	{
+	}
+
+}
+
+abstract class AbstractParentExample
+{
+
+	/**
+	 * @throws RuntimeException
+	 */
+	abstract public function method(): void;
+
+}
+
+class AbstractExample extends AbstractParentExample {
+
+	public function method(): void
+	{
+	}
+
+}
+
+class OverridingAbstractExample extends AbstractExample
+{
+
+	public function method(): void
+	{
+	}
+
+}
+
+class ConcreteParentExample
+{
+
+	/**
+	 * @throws RuntimeException
+	 */
+	public function method(): void
+	{
+		throw new RuntimeException();
+	}
+
+}
+
+class OverridingConcreteExample extends ConcreteParentExample
+{
+
+	public function method(): void
```

Comment:
This method should report unused throws because it's not an implementation of the abstract method. It just overrides the parent method. But I'm still thinking if it would be what we want. Maybe it would be confusing.

Reply:
I thought a lot about it. I believe that both fall in the same use case (LSP). Also, I've asked some guys about it, and they didn't find it confusing.

Reply:
In fact, that was their expectation.

Reply:
So you suggest to merge it as is?

Reply:
Yeah, I believe that who wants the former case also wants the latter.

Reply:
It's not a technical issue, we can toggle this behavior by just [changing this line](https://github.com/pepakriz/phpstan-exception-rules/pull/95/files#diff-ba58f30e348c189bfe738b0e54f3f226R611) to check whether the prototype is abstract.